### PR TITLE
Added secrets scanner, attached to pre-commit, updated readme

### PR DESCRIPTION
**Pull request by ksunandp**
### Changes
Adds https://github.com/redbadger/secrets-scanner -> scanner
Adds to makefile to make use of scanner
Adds scan to pre-commit -> package.json
Adds scanner to .gitignore

#### Trello Ticket / Github Issue Link
https://trello.com/c/ZvEYcY4B/123-issue-599-preventing-publishing-of-secret-information
https://github.com/redbadger/website-honestly/issues/599

#### Description
Using https://github.com/redbadger/secrets-scanner, we are able to add a git hook with pre-git to scan for secrets before any accidental commit of said secrets. Patterns are defined within the added repo but can be added project based locally.

### Test
Review needed

#### Can this PR be merged immediately after testing (eg. rebase, infrastructure update)?
Yes

### Docs and training
For reference:
https://github.com/redbadger/secrets-scanner
https://github.com/awslabs/git-secrets

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -176,6 +176,24 @@ dist/static-site:
 	mkdir -p ./dist/static-site
 	@$(PRINT_OK)
 
+scanner: ## Initialize git secrets in the scanner folder
+	git clone git@github.com:redbadger/secrets-scanner.git scanner
+	@echo ""
+	@echo "*************************************************************"
+	@echo "* Follow the instructions to get added to the setup the scanner:"
```
Comment (dpiatek) [inline Makefile]: `"Follow the instructions below to setup the scanner?"`

File: README.md [hunk 2]
```diff
@@ -34,6 +34,14 @@ make keyrings
 # - "Getting Blackbox"
 # - "Get access to existing blackbox secrets"
 
+# Set up the git secrets to scan for secrets pre-commit. Follow the instructions of this command:
```
Comment (dpiatek) [inline README.md]: `"Set up git-secrets ..."` ?

**Comment by dpiatek** [comment]
Some small copy comments but otherwise looks great!

**Comment by ksunandp** [comment]
Nice catches, changes made.

**Pull request merged**